OmsAgent/Fairfax/omsagent_ff.py

```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
+
+    if other_connection_exists:
+        err_msg = ('This machine is already connected to some other Log ' \
+                   'Analytics workspace, please set ' \
+                   'stopOnMultipleConnections to false in public ' \
+                   'settings or remove this property, so this machine ' \
+                   'can connect to new workspaces, also it means this ' \
+                   'machine will get billed multiple times for each ' \
+                   'workspace it report to. ' \
+                   '(LINUXOMSAGENTEXTENSION_ERROR_MULTIPLECONNECTIONS)')
+        # This exception will get caught by the main method
+        raise OmsAgentUnwantedMultipleConnectionsException(err_msg)
+    else:
+        detect_scom_connection
+
+
+def detect_scom_connection():
+    """
+    If these two conditions are met, then we can assume the
+    VM is monitored
+    by SCOM:
+    1. Port 1270 is open and omiserver is listening on it
+    2. scx certificate is signed by SCOM server
+
+    To determine it check for existence of below two
+    conditions:
+    1. Port 1270 is open and omiserver is listening on it:
+       /etc/omi/conf/omiserver.conf can be parsed to
+       determine it.
+    2. scx certificate is signed by SCOM server: scom cert
+       is present @ /etc/opt/omi/ssl/omi-host-<hostname>.pem
+       (/etc/opt/microsoft/scx/ssl/scx.pem is a softlink to
+       this). If the machine is monitored by SCOM then issuer
+       field of the certificate will have a value like
+       CN=SCX-Certificate/title=<GUID>, DC=<SCOM server hostname>
+       (e.g CN=SCX-Certificate/title=SCX94a1f46d-2ced-4739-9b6a-1f06156ca4ac,
+       DC=NEB-OM-1502733)
+
+    Otherwise, if a scom configuration directory has been
+    created, we assume SCOM is in use
+    """
+    scom_port_open = None # return when determine this is false
+    cert_signed_by_scom = False
+
+    if os.path.exists(OmsAdminPath):
+        # If packages have not been installed yet, this may not exist
+        omsadmin_cmd = '{0} -o'.format(OmsAdminPath)
+        exit_code, output = run_get_output(omsadmin_cmd, False, False)
+        # Guard against older omsadmin.sh versions
+        if ('illegal option' not in output.lower()
+                and 'unknown option' not in output.lower()):
+            if exit_code is 0:
+                hutil_log_info('According to {0}, SCOM port is ' \
+                               'open'.format(omsadmin_cmd))
+                scom_port_open = True
+            elif exit_code is 1:
+                hutil_log_info('According to {0}, SCOM port is not ' \
+                               'open'.format(omsadmin_cmd))
+                return
+
+    # If omsadmin.sh option is not available, use omiconfigeditor
+    if (scom_port_open is None and os.path.exists(OMIConfigEditorPath)
+            and os.path.exists(OMIServerConfPath)):
+        omi_cmd = '{0} httpsport -q 1270 < {1}'.format(OMIConfigEditorPath,
```

Comment:
Move hardcoded 1270 port number to be in global variables.

Reply:
Done